pennylane/devices/qubit/measure.py

```diff
@@ -28,67 +28,92 @@
 
 
 def state_diagonalizing_gates(
-    measurementprocess: StateMeasurement, state: TensorLike
+    measurementprocess: StateMeasurement, state: TensorLike, is_state_batched: bool = False
 ) -> TensorLike:
     """Apply a measurement to state when the measurement process has an observable with diagonalizing gates.
 
     Args:
         measurementprocess (StateMeasurement): measurement to apply to the state
         state (TensorLike): state to apply the measurement to
+        is_state_batched (bool): whether the state is batched or not
 
     Returns:
         TensorLike: the result of the measurement
     """
     for op in measurementprocess.diagonalizing_gates():
-        state = apply_operation(op, state)
+        state = apply_operation(op, state, is_state_batched=is_state_batched)
 
-    total_indices = len(state.shape)
+    total_indices = len(state.shape) - is_state_batched
     wires = Wires(range(total_indices))
+
+    if is_state_batched:
+        return measurementprocess.process_state(math.reshape(state, (state.shape[0], -1)), wires)
+
     return measurementprocess.process_state(math.flatten(state), wires)
 
 
-def csr_dot_products(measurementprocess: ExpectationMP, state: TensorLike) -> TensorLike:
+def csr_dot_products(
+    measurementprocess: ExpectationMP, state: TensorLike, is_state_batched: bool = False
+) -> TensorLike:
     """Measure the expectation value of an observable using dot products between ``scipy.csr_matrix``
     representations.
 
     Args:
         measurementprocess (ExpectationMP): measurement process to apply to the state
         state (TensorLike): the state to measure
+        is_state_batched (bool): whether the state is batched or not
 
     Returns:
         TensorLike: the result of the measurement
     """
-    total_wires = len(state.shape)
+    total_wires = len(state.shape) - is_state_batched
     Hmat = measurementprocess.obs.sparse_matrix(wire_order=list(range(total_wires)))
-    state = math.toarray(state).flatten()
 
-    # Find the expectation value using the <\psi|H|\psi> matrix contraction
-    bra = csr_matrix(math.conj(state))
-    ket = csr_matrix(state[..., None])
-    new_ket = csr_matrix.dot(Hmat, ket)
-    res = csr_matrix.dot(bra, new_ket).toarray()[0]
+    if is_state_batched:
+        state = math.toarray(state).reshape(math.shape(state)[0], -1)
+
+        bra = csr_matrix(math.conj(state))
+        ket = csr_matrix(state)
+        new_bra = bra.dot(Hmat)
+        res = new_bra.multiply(ket).sum(axis=1)
```

Comment:
```suggestion
        res = new_bra.multiply(ket).sum(axis=1).getA()
```

the `csr_matrix.sum` method returns a `numpy.matrix`, when we want a `numpy.ndarray`.  `matrix` can only be 2D, so therefore it wasn't actually squeezing.